piAIOComm.c

```diff
@@ -318,7 +318,9 @@ u32 piAIOComm_sendCyclicTelegram(u8 devnum)
 
 	ret = pibridge_req_io(addr, IOP_TYP1_CMD_DATA, snd_buf,
 			      AIO_OUTPUT_DATA_LEN, rcv_buf, AIO_INPUT_DATA_LEN);
-	if (ret) {
+	if (ret != AIO_INPUT_DATA_LEN) {
+		if (ret > 0)
+			ret = -EIO;
 		pr_warn_ratelimited("addr %2d: io communication failed with AIO %d\n",
 			addr, ret);
 		return 1;
```

Comment:
Hm, I'm confused because the error path unconditionally returns `1`, so changing `ret` to `-EIO` only affects the warning message. Is that intentional? There's arguably not much of a benefit for the warning message to contain -EIO instead of a positive integer...

Reply:
Yes, thats intentional. The only reason for the for the adjustment of "ret" was to have an error value in the following message instead of the number of received bytes